File: benchmarks/vm/04_filter_sum.rb
t0 = Process.clock_gettime(Process::CLOCK_MONOTONIC)
nums = (0...10000).to_a
sum = 0
nums.each { |n| sum += n if n.even? }
puts "sum = #{sum}"
puts "BENCH_RESULT: #{((Process.clock_gettime(Process::CLOCK_MONOTONIC) - t0) * 1000).to_i} ms"

File: benchmarks/vm/04_filter_sum.py
import time
t0 = time.monotonic()
nums = list(range(10000))
total = sum(n for n in nums if n % 2 == 0)
print(f"sum = {total}")
print(f"BENCH_RESULT: {int((time.monotonic() - t0) * 1000)} ms")
